### ISSUE TITLE
<<=>>> can crash if RHS is a List in a Scalar container

### ISSUE THREAD

**Issue by jnthn** [rakudo/rakudo#2674]
## The Problem

This test:

```
use Test;
is-deeply { a => (1,2,3) }.map({ .key <<=>>> .value }).list, ((:1a, :2a, :3a),)
```

Dies.

## Expected Behavior

It passes.

## Actual Behavior

Dies with:

```
P6opaque: no such attribute '$!reified' on type List in a Scalar when trying to bind a value
  in block <unit> at -e line 1
```

This is inside of method `infix` in `src/core/Hyper.pm6`, and due to an `nqp::create` that doesn't first `nqp::decont` its argument. Using `.CREATE` would have been safer.

## Environment
* Operating system: Linux, OSX
* Compiler version (`perl6 -v`): Present in 2018.12 release, still in 2018.12-268-g6fe5377

**Comment by lizmat** [comment]
Fixed with df96fa0f34

**Comment by jnthn** [comment]
Thanks for the fix; test added.
